src/providers/wfs/qgswfsprovider.cpp

```diff
@@ -1657,7 +1657,15 @@ QDomElement QgsWFSProvider::createTransactionElement( QDomDocument &doc ) const
     describeFeatureTypeURL.setQuery( query );
   }
   QUrlQuery query( describeFeatureTypeURL );
-  query.addQueryItem( QStringLiteral( "VERSION" ), QStringLiteral( "1.0.0" ) );
+  // only 1.1.0 and 1.0.0 are supported
+  if ( WfsVersion == QStringLiteral( "1.1.0" ) )
+  {
+    query.addQueryItem( QStringLiteral( "VERSION" ), WfsVersion );
```

Comment:
Is the `if/else` necessary? Why cannot we just always do `query.addQueryItem( QStringLiteral( "VERSION" ), WfsVersion );` ?

Reply:
My understanding is that it matches [this logic](https://github.com/qgis/QGIS/blob/13e4969817e990629748a3342938df0c772e14c4/src/providers/wfs/qgswfsprovider.cpp#L1639-L1646) above. Though one could make an argument for storing the "effective version" in a variable as part of that first check and using it thereafter...

Reply:
Ah, I see now it's from the line 1639 above. Maybe you could set a version string there and avoid code duplication?

Reply:
Code updated